pkg/workflows/wasm/host/module.go

```diff
@@ -326,7 +338,18 @@ func (m *Module) Close() {
 	m.module.Close()
 }
 
-func (m *Module) Run(request *wasmpb.Request) (*wasmpb.Response, error) {
+func (m *Module) Run(ctx context.Context, request *wasmpb.Request) (*wasmpb.Response, error) {
+	if request == nil {
+		return nil, fmt.Errorf("invalid request: can't be nil")
+	}
+
+	if request.Id == "" {
+		return nil, fmt.Errorf("invalid request: can't be empty")
+	}
+
+	// we add the request context to the store to make it available to the Fetch fn
+	m.requestStore.add(request.Id, &RequestData{ctx: ctx})
```

Comment:
I'm surprised linting hasn't complained here -- you need to check that we didn't error here